### DISCUSSION TITLE
Generation Vs Error Plot

### DISCUSSION THREAD

**Discussion by p-suresh-kumar** [ahmedfgad/GeneticAlgorithmPython#36] [General]
Dear GAD, 
is there any function to plot generation vs error plot,
or how to find the error at each generation?

**Comment by ahmedfgad** [comment]
There is only function to plot the generation vs fitness.

If you want to see plot the graph of the error not the fitness, then here is how you can do it for the `pygad.kerasga` module:

The `best_solutions_fitness` attribute of instances from the `pygad.GA` class holds the best fitness value in each generation. You can convert the fitness values to errors as follows:
```python
best_solutions_error = 1.0 / numpy.array(ga_instance.best_solutions_fitness)
```

Then show the graph:

```python
matplotlib.pyplot.plot(best_solutions_error)
matplotlib.pyplot.show()
```

That should work. Please let me know if you have any doubts.
